File: src/test/java/celtric/katas/game_of_life_2017_05/tests/acceptance/patterns/SpaceshipsTest.java
package celtric.katas.game_of_life_2017_05.tests.acceptance.patterns;

import celtric.katas.game_of_life_2017_05.Game;
import celtric.katas.game_of_life_2017_05.tests.acceptance.AcceptanceTestCase;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public final class SpaceshipsTest extends AcceptanceTestCase {

    @Test
    public void glider() {
        Game game = game(4,
            0, A, 0, 0,
            0, 0, A, 0,
            A, A, A, 0,
            0, 0, 0, 0);

        assertEquals(game(4,
            0, 0, 0, 0,
            A, 0, A, 0,
            0, A, A, 0,
            0, A, 0, 0), game.evolve());

        assertEquals(game(4,
            0, 0, 0, 0,
            0, 0, A, 0,
            A, 0, A, 0,
            0, A, A, 0), game.evolve(2));

        assertEquals(game(4,
            0, 0, 0, 0,
            0, A, 0, 0,
            0, 0, A, A,
            0, A, A, 0), game.evolve(3));

        assertEquals(game(4,
            0, 0, 0, 0,
            0, 0, A, 0,
            0, 0, 0, A,
            0, A, A, A), game.evolve(4));
    }
}

File: src/test/kotlin/celtric/katas/game_of_life_2017_05/tests/acceptance/patterns/SpaceshipsTest.kt
package celtric.katas.game_of_life_2017_05.tests.acceptance.patterns

import celtric.katas.game_of_life_2017_05.tests.acceptance.AcceptanceTestCase
import org.junit.Assert.assertEquals
import org.junit.Test

class SpaceshipsTest : AcceptanceTestCase() {

    @Test
    fun glider() {
        val game = game(
                4,
                0, a, 0, 0,
                0, 0, a, 0,
                a, a, a, 0,
                0, 0, 0, 0)

        assertEquals(
                game(
                        4,
                        0, 0, 0, 0,
                        a, 0, a, 0,
                        0, a, a, 0,
                        0, a, 0, 0), game.evolve())

        assertEquals(
                game(
                        4,
                        0, 0, 0, 0,
                        0, 0, a, 0,
                        a, 0, a, 0,
                        0, a, a, 0), game.evolve(2))

        assertEquals(
                game(
                        4,
                        0, 0, 0, 0,
                        0, a, 0, 0,
                        0, 0, a, a,
                        0, a, a, 0), game.evolve(3))

        assertEquals(
                game(
                        4,
                        0, 0, 0, 0,
                        0, 0, a, 0,
                        0, 0, 0, a,
                        0, a, a, a), game.evolve(4))
    }
}
